events/permissions.py

```diff
@@ -113,6 +113,51 @@ def has_permission(self, request, view):
 )
 
 
+class OrganizationAddRemoveUsersPermission(BasePermission):
+    @staticmethod
+    def _has_added_or_removed_users(request_data, obj):
+        if "admin_users" in request_data and set(request_data["admin_users"]) != set(
+            obj.admin_users.values_list("username", flat=True)
+        ):
+            return True
+
+        if "registration_admin_users" in request_data and set(
+            request_data["registration_admin_users"]
+        ) != set(obj.registration_admin_users.values_list("username", flat=True)):
+            return True
+
+        if "financial_admin_users" in request_data and set(
+            request_data["financial_admin_users"]
+        ) != set(obj.financial_admin_users.values_list("username", flat=True)):
+            return True
+
+        if "regular_users" in request_data and set(
+            request_data["regular_users"]
+        ) != set(obj.regular_users.values_list("username", flat=True)):
+            return True
+
+        return False
+
+    def has_permission(self, request, view):
+        if request.method == "POST" and (
+            request.data.get("admin_users")
+            or request.data.get("registration_admin_users")
+            or request.data.get("financial_admin_users")
+            or request.data.get("regular_users")
+        ):
+            return request.user.is_authenticated and request.user.is_superuser
+
+        return True
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in ("PUT", "PATCH") and self._has_added_or_removed_users(
+            request.data, obj
+        ):
+            return request.user.is_authenticated and request.user.is_superuser
+
+        return True
```

Comment:
I combined this permission with `DataSourceOrganizationEditPermission` and since that existing permission didn't actually have anything to do with data sources, I renamed it to `OrganizationEditPermission`.